src/game/rabble.ts

```diff
@@ -3,23 +3,72 @@ import {
   createTileBag,
   shuffleTiles,
   drawTiles,
-  checkForPlayTilesInRack,
   pullPlayTilesFromRack,
   exchangeTiles,
 } from "./tileBag";
-import { hasCenterSquare, generateBoard, isFirstPlay } from "./board";
+import {
+  hasCenterSquare,
+  generateBoard,
+  isFirstPlay,
+  // consolePrintBoard,
+} from "./board";
 import {
   allTilesFromSquares,
   playTilesFromSquares,
   finalizePlayOnBoard,
   adjacentToAWord,
   playDirection,
   copyPlaySquaresToBoard,
-  removePlayFromBoard,
   allSquaresInWord,
 } from "./play";
 import { checkForInvalidWords, scoreForValidWords } from "./score";
 
+const currentPlayInfo = (
+  override: Partial<CurrentPlayInfo>
+): CurrentPlayInfo => ({
+  invalidReason: "",
+  valid: false,
+  tilesLaid: [],
+  played: false,
+  score: 0,
+  ...override,
+});
+
+const playIsInvalid = (
```

Comment:
This function is a little weird, was thinkin-
1) rename it "checkValidity" and also set valid to true if there's no problems
2) rename it "findProblems" or something like that, which is what it does now

Could probably pass in currentPlay as a variable to make it more functional-style